app/controllers/api/api_controller.rb

```diff
@@ -15,6 +16,12 @@ class Api::ApiController < ActionController::Base
 
   protected
 
+  def oauth_or_jwt
+    @current_user = @current_user || current_token&.user
+  end
+  def current_user=(user)
```

Comment:
(This is strange – `x=` methods should be setters, not equality checkers. Why not just access `current_user` directly where you need to check it?)